src/consensus/pbft_manager.cpp

```diff
@@ -1852,7 +1852,7 @@ void PbftManager::syncBlockQueuePush(SyncBlock &&block, dev::p2p::NodeID const &
 }
 
 void PbftManager::clearSyncBlockQueue() {
-  sync_period_ = 0;
+  sync_period_ = pbft_chain_->getPbftChainSize();
```

Comment:
I believe there can be a race condition between syncBlockQueuePush and clearSyncBlockQueue. Scenario:
Queue already contains block period 7,8 and 9.
At the same time one thread enters syncBlockQueuePush and another clearSyncBlockQueue
First thread successfully skips if (period != pbftSyncingPeriod() + 1) 
After that second thread sets sync_peiod to 0 and clears the queue
First thread then pushes belock period 10 in the queue which is invalid since periods 7,8 and 9 were deleted
Solution would be putting everything in these two methods under the same lock

Reply:
@mfrankovi yop I think you are right.